pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/StateChangeListeners.java

```diff
@@ -54,6 +54,17 @@ public <T> CompletableFuture<T> notifyOnCompletion(CompletableFuture<T> future,
         return future.whenComplete((r, ex) -> notify(serviceUnit, data, ex));
     }
 
+    public void notifyOnArrival(String serviceUnit, ServiceUnitStateData data) {
```

Comment:
Sorry. I realized that we can call this `notifyOnArrival` in the `notifyOnCompletion` at line 53. Then, we don't need to call `notifyOnArrival` in the ServiceUnitStateChannelImpl explicitly. (this means we can make  notifyOnArrival a private func)